web/styles/search.css

```diff
@@ -2,150 +2,232 @@
-        overflow: hidden;
-        margin-top: 0;
-        margin-bottom: 0;
-        width: calc(100% - 2px);
-        position: absolute;
+    .search_icon {
+        text-decoration: none;
+        padding: 0 7px;
+        font-size: 17px;
+        border: none;
+        background-color: transparent;
+        color: hsl(0deg 0% 0%);
+        opacity: 0.5;
+        cursor: default;
 
-        @media (width < $md_min) {
-            width: calc(100% - 42px);
+        &:active,
+        &:focus {
+            outline: none;
         }
 
-        .search_close_button {
-            display: inline;
-            margin-right: 5px;
+        &:disabled {
+            visibility: hidden;
         }
+    }
+
+    .search_close_button {
+        width: 28px;
+        height: 100%;
+        background: none;
+        border: none;
+        opacity: 0.5;
+        line-height: 0;
 
-        .zulip-icon-close {
-            font-size: 15px;
+        &:not(:focus-visible) {
+            outline: none;
+        }
+
+        &:hover {
+            opacity: 1;
+        }
+
+        &:disabled {
+            visibility: hidden;
         }
     }
 
-    .typeahead.dropdown-menu {
-        /* Match the typeahead's width to its parent container. */
+    .zulip-icon-close {
+        font-size: 13px;
+    }
+
+    .navbar-search:not(.expanded) {
         right: 0;
 
+        .search_close_button {
+            display: none;
+        }
+
+        #searchbox-input-container {
+            width: 150px;
+        }
+
+        .search-input {
+            border-radius: 5px;
+            color: var(--color-text-search);
+
+            box-shadow: none;
+            cursor: pointer;
+        }
+
+        &:hover {
+            cursor: pointer;
+
+            @media (width >= $md_min) {
+                box-shadow:
+                    0 4px 20px var(--color-search-box-hover-shadow),
+                    0 1px 5px var(--color-search-box-hover-shadow);
+            }
+
+            .search_icon {
+                cursor: pointer;
+            }
+        }
+
         @media (width < $md_min) {
-            margin-left: 40px;
+            #searchbox-input-container {
+                width: 40px;
+            }
+
+            .search-input {
+                opacity: 0;
+            }
+        }
+
+        @media (width < $sm_min) {
+            #searchbox-input-container .search_icon {
+                font-size: 18px;
+            }
         }
     }
 
-    .input-append {
-        position: relative;
+    .navbar-search.expanded {
+        color: hsl(0deg 0% 0% / 100%);
         width: 100%;
-        width: max-content;
-        min-width: 100%;
-
-        .zulip-icon-search {
-            padding: 0;
-            font-size: 20px;
-            position: absolute;
-            left: 10px;
-            top: 10px;
-            z-index: 5;
-
-            @media (width < $sm_min) {
-                top: 5px;
+        box-shadow:
+            0 4px 20px hsl(0deg 0% 0% / 25%),
```

Comment:
Are these colors different from the box-shadows above that have variables assigned to them? I'm guessing so, but there's another case where variables would be an asset on wrangling all of these colors.

Reply:
Yeah this one is shadow around the whole search situation (input field and typeahead) when it's open, to create some visual separation from the rest of the page.

Can you recommend what to call the colors? The first is an bigger shadow and the second is a tighter one. I made them blue and red below to show what they do:

<img width="850" alt="image" src="https://github.com/zulip/zulip/assets/5634097/8a8d5871-5b0f-4527-9529-b2dcb5a9e1e6">

Reply:
Updated to

```css

        box-shadow:
            0 4px 20px var(--color-search-shadow-wide),
            0 1px 5px var(--color-search-shadow-tight);

```